airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -381,175 +381,302 @@ static Optional<ContainerOrchestratorConfig> getContainerOrchestratorConfig(fina
+    if (configs.shouldRunCheckConnectionWorkflows()) {
+      registerCheckConnection(factory);
+    }
 
-    final SecretsHydrator secretsHydrator = SecretPersistence.getSecretsHydrator(configsDslContext, configs);
+    if (configs.shouldRunDiscoverWorkflows()) {
+      registerDiscover(factory);
+    }
 
-    if (configs.getWorkerEnvironment().equals(WorkerEnvironment.KUBERNETES)) {
-      KubePortManagerSingleton.init(configs.getTemporalWorkerPorts());
+    if (configs.shouldRunSyncWorkflows()) {
+      registerSync(factory);
     }
 
-    final Database configDatabase = new Database(configsDslContext);
-    final FeatureFlags featureFlags = new EnvVariableFeatureFlags();
-    final JsonSecretsProcessor jsonSecretsProcessor = JsonSecretsProcessor.builder()
-        .maskSecrets(!featureFlags.exposeSecretsInExport())
-        .copySecrets(false)
-        .build();
-    final ConfigPersistence configPersistence = DatabaseConfigPersistence.createWithValidation(configDatabase, jsonSecretsProcessor);
-    final ConfigRepository configRepository = new ConfigRepository(configPersistence, configDatabase);
+    if (configs.shouldRunConnectionManagerWorkflows()) {
+      registerConnectionManager(factory);
+    }
 
-    final Database jobDatabase = new Database(jobsDslContext);
+    factory.start();
+  }
 
-    final JobPersistence jobPersistence = new DefaultJobPersistence(jobDatabase);
-    final StatePersistence statePersistence = new StatePersistence(configDatabase);
-    final DefaultJobCreator jobCreator = new DefaultJobCreator(
-        jobPersistence,
-        defaultWorkerConfigs.getResourceRequirements(),
-        statePersistence);
-
-    TrackingClientSingleton.initialize(
-        configs.getTrackingStrategy(),
-        new Deployment(configs.getDeploymentMode(), jobPersistence.getDeployment().orElseThrow(), configs.getWorkerEnvironment()),
-        configs.getAirbyteRole(),
-        configs.getAirbyteVersion(),
-        configRepository);
-    final TrackingClient trackingClient = TrackingClientSingleton.get();
-    final SyncJobFactory jobFactory = new DefaultSyncJobFactory(
-        configs.connectorSpecificResourceDefaultsEnabled(),
-        jobCreator,
-        configRepository,
-        new OAuthConfigSupplier(configRepository, trackingClient));
-
-    final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalService();
-
-    final WorkflowClient workflowClient = TemporalUtils.createWorkflowClient(temporalService, TemporalUtils.getNamespace());
-    final StreamResetPersistence streamResetPersistence = new StreamResetPersistence(configDatabase);
-
-    final TemporalClient temporalClient = new TemporalClient(workflowClient, configs.getWorkspaceRoot(), temporalService, streamResetPersistence);
-    TemporalUtils.configureTemporalNamespace(temporalService);
+  private static void initializeCommonDependencies() {
+    LOGGER.debug("Initializing common worker dependencies.");
```

Comment:
nit: can we turn this into an INFO? I think it will be useful for operational work.